osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/misc/GetActualPowerQualityCommandExecutorTest.java

```diff
@@ -33,84 +46,75 @@
 import org.opensmartgridplatform.dto.valueobjects.smartmetering.ActualPowerQualityResponseDto;
-import org.opensmartgridplatform.dto.valueobjects.smartmetering.DlmsUnitTypeDto;
 import org.opensmartgridplatform.dto.valueobjects.smartmetering.PowerQualityObjectDto;
 import org.opensmartgridplatform.dto.valueobjects.smartmetering.PowerQualityValueDto;
 import org.opensmartgridplatform.shared.infra.jms.MessageMetadata;
 
 @ExtendWith(MockitoExtension.class)
-public class GetActualPowerQualityCommandExecutorTest {
+class GetActualPowerQualityCommandExecutorTest {
   private static final int CLASS_ID_DATA = 1;
   private static final int CLASS_ID_REGISTER = 3;
   private static final int CLASS_ID_CLOCK = 8;
+  private static final int ATTRIBUTE_ID_VALUE = 2;
+  private static final int ATTRIBUTE_ID_SCALER_UNIT = 3;
+  private static final String PROTOCOL_NAME = "SMR";
+  private static final String PROTOCOL_VERSION = "5.0.0";
+  private static final MessageMetadata MESSAGE_METADATA =
+      MessageMetadata.newBuilder().withCorrelationUid("123456").build();
 
   @Spy private DlmsHelper dlmsHelper;
 
+  @Mock private ObjectConfigService objectConfigService;
+
   @Mock private DlmsDevice dlmsDevice;
 
   @Mock private DlmsConnectionManager conn;
 
   @Mock private DlmsMessageListener dlmsMessageListener;
 
-  private ActualPowerQualityRequestDto actualPowerQualityRequestDto;
-  private MessageMetadata messageMetadata;
-
-  @BeforeEach
-  public void before() throws ProtocolAdapterException, IOException {
-    this.messageMetadata = MessageMetadata.newBuilder().withCorrelationUid("123456").build();
-
-    when(this.conn.getDlmsMessageListener()).thenReturn(this.dlmsMessageListener);
-  }
-
-  @Test
-  void testRetrievalPublic() throws ProtocolAdapterException {
-    when(this.dlmsDevice.isPolyphase()).thenReturn(true);
-    this.executeAndAssert("PUBLIC", GetActualPowerQualityCommandExecutor.getMetadatasPublic());
-  }
+  @Captor ArgumentCaptor<AttributeAddress> attributeAddressArgumentCaptor;
 
-  @Test
-  void testRetrievalPrivate() throws ProtocolAdapterException {
-    when(this.dlmsDevice.isPolyphase()).thenReturn(true);
-    this.executeAndAssert("PRIVATE", GetActualPowerQualityCommandExecutor.getMetadatasPrivate());
-  }
+  private ActualPowerQualityRequestDto actualPowerQualityRequestDto;
 
-  @Test
-  void testRetrievalPublicSinglePhase() throws ProtocolAdapterException {
-    when(this.dlmsDevice.isPolyphase()).thenReturn(false);
-    this.executeAndAssert(
-        "PUBLIC",
-        GetActualPowerQualityCommandExecutor.getMetadatasPublic().stream()
-            .filter(metadata -> !metadata.isPolyphaseOnly())
-            .collect(Collectors.toList()));
+  @ParameterizedTest
+  @EnumSource(PowerQualityProfile.class)
+  void testRetrievalPolyPhase(final PowerQualityProfile profile)
+      throws ProtocolAdapterException, ObjectConfigException {
+    this.executeAndAssert(profile.name(), true);
```

Comment:
Waarom hier geen tweede EnumSource (MeterType) zodat deze test en onderstaande voor SinglePhase kunnn worden samengevoegd?
Alle aangeroepen methodes kunnen MeterType ontvangen i.p.v. boolean.
Bovendien kan de terugvertaling van boolean naar MeterType dan achterwege blijven
`private List<MeterType> getMeterTypes(final boolean polyphase)`

Reply:
Changed boolean to MeterType